infra/main.bicep

```diff
@@ -8,75 +8,86 @@ param environmentName string
+var tags = { 'azd-env-name': environmentName }
+
 
 // Organize resources in a resource group 
-resource group 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
   name: 'rg-${environmentName}'
   location: location
   tags: tags
 }
 
-module openAI 'br/public:ai/cognitiveservices:1.1.1' = {
+module openAi 'br/public:avm/res/cognitive-services/account:0.5.3' = {
   name: 'openai'
-  scope: group
+  scope: rg
   params: {
     tags: tags
-    skuName: 'S0'
     kind: 'OpenAI'
     name: 'openai-${resourceToken}'
     location: location
+    disableLocalAuth: false
+    publicNetworkAccess: 'Enabled'
     deployments: [
       {
-        name: 'Gpt35Turbo_0301'
-        properties: {
-          model: {
-            format: 'OpenAI'
-            name: 'gpt-35-turbo'
-            version: '0301'
-          }
+        name: azureOpenAIModelName
+        model: {
+          format: 'OpenAI'
+          name: azureOpenAIModel
+          version: azureOpenAIModelVersion
         }
-      }
-      {
-        name: 'TextEmbeddingAda002_1'
-        properties: {
-          model: {
-            format: 'OpenAI'
-            name: 'text-embedding-ada-002'
-            version: '2'
-          }
+        sku: {
+          capacity: 10
+          name: 'Standard'
         }
       }
       {
-        name: 'Gpt35Turbo_16k'
-        properties: {
-          model: {
-            format: 'OpenAI'
-            name: 'gpt-35-turbo-16k'
-            version: '0613'
-          }
+        name: azureOpenAIEmbeddingModelName
+        model: {
+          format: 'OpenAI'
+          name: azureOpenAIEmbeddingModel
+          version: azureOpenAIEmbeddingModelVersion
+        }
+        sku: {
+          capacity: 10
+          name: 'Standard'
         }
       }
     ]
   }
 }
 
-module searchService 'br/public:search/search-service:1.0.2' = {
-  name: 'search-service'
-  scope: group
-  params: {
-    name: 'search-${resourceToken}'
-    location: location
-    tags: tags
-    sku: {
-      name: 'free'
-    }
-  }
-}
-
 output AZURE_LOCATION string = location
-output AZURE_RESOURCE_GROUP string = group.name
-output AZURE_TENANT_ID string = tenant().tenantId
-output AZURE_OPENAI_ENDPOINT string = openAI.outputs.endpoint
-output AZURE_SEARCH_ENDPOINT string = searchService.outputs.endpoint
+output AZURE_RESOURCE_GROUP string = rg.name
+output OPENAI_HOST string = openAiHost
+output AZURE_OPENAI_ENDPOINT string = (openAiHost == 'azure') ? openAi.outputs.endpoint: ''
+output AZURE_OPENAI_SERVICE string = (openAiHost == 'azure') ? openAi.outputs.name: ''
+output AZURE_OPENAI_MODEL string = (openAiHost == 'azure') ? azureOpenAIModelName: ''
+output AZURE_OPENAI_EMBEDDING_MODEL string = (openAiHost == 'azure') ? azureOpenAIEmbeddingModelName: ''
+
+output OPENAI_API_KEY string = (openAiHost == 'openai') ? openAiApiKey : ''
```

Comment:
We shouldn't be outputting the key in bicep as that will be in the deployment.  If the user wants to set it, then they they will to run azd env set.